### PR TITLE
Update Cloudflare Domain Configuration

### LINKED ISSUE #2230: Adding HTTPS For Free With Cloudflare Doc Update
Re: [Adding HTTPS For Free With Cloudflare](https://pantheon.io/docs/guides/cloudflare-enable-https)

Priority (Low‚ Medium‚ High): Medium

## Issue Description:

While the guide already recommends Full mode, we've made an improvement where Flexible mode will not work at all now for new sites, so let's make it explicit to not use Flexible mode.

## Suggested Resolution 

- add wording to not use Flexible.
- update screenshot where Full ix circled to include red X for Flexible

### PR DESCRIPTION
**Pull request by ari-gold**
Close #2230 

## Effect
PR includes the following changes:
- Add section on when to use Cloudflare as a CDN
- Add screenshots
- When using Cloudflare, always use a CNAME
- Delete cloudflare guide and redirect to dns provider doc

## Remaining Work
- [x] Review from @rachelwhitton and optionally @davidstrauss or @joshkoenig 
- [x] Remove existing Cloudflare guide and add redirect to this doc

### CODE REVIEW

**Comment by rachelwhitton** [comment]
+1 Looks great - I made some heavy edits, can you review? If it looks good we're ready to deploy on your mark @ari-gold. I'll create the redirect when we ship.

File: source/_docs/free-https.md [hunk 1]
```diff
@@ -104,7 +104,7 @@ All modern browsers and operating systems are supported. For details, see the **
 ### What about CloudFlare or other CDNs?
 Many customers currently take advantage of CloudFlare's awesome Universal SSL offering to get free HTTPS service for their website. If you are _just_ using CloudFlare for the HTTPS service, you can switch to the Global CDN and get an upgrade in performance and SSLLabs score.
 
-However, customers using CloudFlare's WAF tools or other features may want to keep CloudFlare in their stack. There are no known issues with layering CloudFlare and the Global CDN together, but we strongly recommend you enforce HTTPS in Cloudflare and within WordPress or Drupal to avoid mixed content.
+However, customers using CloudFlare's WAF tools or other features may want to keep CloudFlare in their stack. We still recommend upgrading your Pantheon site to the Global CDN as this will improve cache hitrates and performance. There are no known issues with layering CloudFlare and the Global CDN together. For details, see [Cloudflare Domain Configuration](/docs/cloudflare/).
```
Comment (ari-gold) [inline source/_docs/free-https.md]: CloudFlare >> Cloudflare

File: source/_docs/dns-providers/cloudflare.md [hunk 2]
```diff
@@ -2,58 +2,74 @@
...
 5. Select desired Time to Live (TTL).
 
-    {% include("ttl.twig") %}
+  {% include("ttl.twig") %}
 
 6. Disable Cloudflare's CDN by clicking the cloud icon (should be gray, not orange).
-6. Click **Add Record**.
+7. Click **Add Record**.
+
+  ![Example DNS only](/source/docs/assets/images/cloudflare-dns-only.png)
```
Comment (ari-gold) [inline source/_docs/dns-providers/cloudflare.md]: Could be nice to put a border and padding around some or all of these the screenshots.

**Comment by davidstrauss** [comment]
+1 on CNAMES for Cloudflare.

**Pull request merged**